python/sparkdl/transformers/tf_tensor.py

```diff
@@ -62,91 +65,84 @@ def setParams(self, tfInputGraph=None, inputMapping=None, outputMapping=None, tf
+        # Maps names of input tensors in our original graph to outputs of the injected-op subgraph
+        input_map = {}
         with injected_op_subgraph.as_default():
-            # Iterate through the input tensors of the original graph
-            for _, orig_tensor_name in self.getInputMapping():
-                orig_tensor = tfx.get_tensor(orig_tensor_name, user_graph)
-                # Create placeholder with same shape as original input tensor, but that accepts
-                # float64 input from Spark.
-                spark_placeholder = tf.placeholder(tf.float64, shape=orig_tensor.shape,
-                                             name=tfx.op_name(orig_tensor_name))
-                # If the original tensor was of type float64, just pass through the Spark input
-                if orig_tensor.dtype == tf.float64:
-                    output_tensor = tf.identity(spark_placeholder)
-                # Otherwise, cast the Spark input to the datatype of the original tensor
-                else:
-                    output_tensor = tf.cast(spark_placeholder, dtype=orig_tensor.dtype)
-                input_names.append(spark_placeholder.name)
-                output_names.append(output_tensor.name)
-
-        # Concatenate our injected-op subgraph with the original graph and return the result.
-        injected_op_subgraph_fn = GraphFunction(graph_def=injected_op_subgraph.as_graph_def(),
-                                             input_names=input_names, output_names=output_names)
-        return GraphFunction.fromList([("sparkdl_ops", injected_op_subgraph_fn),
-                                      ("user_graph", user_graph_fn)])
+            with tf.name_scope(self.SPARKDL_OP_SCOPE):
+                for _, orig_tensor_name in self.getInputMapping():
+                    orig_tensor = tfx.get_tensor(orig_tensor_name, user_graph)
+                    # Create placeholder with same shape as original input tensor, but that accepts
+                    # float64 input from Spark.
+                    spark_placeholder = tf.placeholder(tf.float64, shape=orig_tensor.shape,
+                                                       name=tfx.op_name(orig_tensor_name))
+                    # If the original tensor was of type float64, just pass through the Spark input
+                    if orig_tensor.dtype == tf.float64:
+                        input_map[orig_tensor_name] = spark_placeholder
+                    # Otherwise, cast the Spark input to the datatype of the original tensor
+                    else:
+                        input_map[orig_tensor_name] = tf.cast(spark_placeholder,
+                                                              dtype=orig_tensor.dtype)
+            tf.import_graph_def(graph_def=user_graph_def, input_map=input_map, name="")
+        return injected_op_subgraph.as_graph_def(add_shapes=True)
 
     def _optimize_for_inference(self):
         """ Optimize the graph for inference """
```

Comment:
not sure this comment helps since it's basically the name of the function - i'd just remove it